pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/AdminResource.java

```diff
@@ -789,4 +796,69 @@ protected void validatePersistencePolicies(PersistencePolicies persistence) {
+                                                                                       String topic) {
+        CompletableFuture<ManagedLedgerInternalStats> statFuture = new CompletableFuture<>();
+        ManagedLedgerInternalStats stats = new ManagedLedgerInternalStats();
+
+        ManagedLedgerImpl ml = (ManagedLedgerImpl) ledger;
+        stats.entriesAddedCounter = ml.getEntriesAddedCounter();
+        stats.numberOfEntries = ml.getNumberOfEntries();
+        stats.totalSize = ml.getTotalSize();
+        stats.currentLedgerEntries = ml.getCurrentLedgerEntries();
+        stats.currentLedgerSize = ml.getCurrentLedgerSize();
+        stats.lastLedgerCreatedTimestamp = DateFormatter.format(ml.getLastLedgerCreatedTimestamp());
+        if (ml.getLastLedgerCreationFailureTimestamp() != 0) {
+            stats.lastLedgerCreationFailureTimestamp = DateFormatter.format(ml.getLastLedgerCreationFailureTimestamp());
+        }
+
+        stats.waitingCursorsCount = ml.getWaitingCursorsCount();
+        stats.pendingAddEntriesCount = ml.getPendingAddEntriesCount();
+
+        stats.lastConfirmedEntry = ml.getLastConfirmedEntry().toString();
+        stats.state = ml.getState();
+
+        stats.ledgers = Lists.newArrayList();
+        ml.getLedgersInfo().forEach((id, li) -> {
+            ManagedLedgerInternalStats.LedgerInfo info = new PersistentTopicInternalStats.LedgerInfo();
+            info.ledgerId = li.getLedgerId();
+            info.entries = li.getEntries();
+            info.size = li.getSize();
+            info.offloaded = li.hasOffloadContext() && li.getOffloadContext().getComplete();
+            stats.ledgers.add(info);
+            if (includeLedgerMetadata) {
+                ml.getLedgerMetadata(li.getLedgerId()).handle((lMetadata, ex) -> {
+                    if (ex == null) {
+                        info.metadata = lMetadata;
+                    }
+                    return null;
+                });
+            }
+
+            stats.cursors = Maps.newTreeMap();
+            ml.getCursors().forEach(c -> {
+                ManagedCursorImpl cursor = (ManagedCursorImpl) c;
+                PersistentTopicInternalStats.CursorStats cs = new PersistentTopicInternalStats.CursorStats();
+                cs.markDeletePosition = cursor.getMarkDeletedPosition().toString();
+                cs.readPosition = cursor.getReadPosition().toString();
+                cs.waitingReadOp = cursor.hasPendingReadRequest();
+                cs.pendingReadOps = cursor.getPendingReadOpsCount();
+                cs.messagesConsumedCounter = cursor.getMessagesConsumedCounter();
+                cs.cursorLedger = cursor.getCursorLedger();
+                cs.cursorLedgerLastEntry = cursor.getCursorLedgerLastEntry();
+                cs.individuallyDeletedMessages = cursor.getIndividuallyDeletedMessages();
+                cs.lastLedgerSwitchTimestamp = DateFormatter.format(cursor.getLastLedgerSwitchTimestamp());
```

Comment:
can this getLastLedgerSwitchTimestamp  return null ?

Reply:
This method does not seem to be used anymore, I forgot to delete it.